tests/OSPSuite.Core.Tests/Domain/ParameterValuesCacheSpecs.cs

```diff
@@ -0,0 +1,351 @@
+﻿using System.Linq;
+using FakeItEasy;
+using OSPSuite.BDDHelper;
+using OSPSuite.BDDHelper.Extensions;
+using OSPSuite.Core.Domain.Populations;
+using OSPSuite.Helpers;
+
+namespace OSPSuite.Core.Domain
+{
+   public abstract class concern_for_ParameterValuesCache : ContextSpecification<ParameterValuesCache>
+   {
+      protected override void Context()
+      {
+         sut = new ParameterValuesCache();
+      }
+   }
+
+   public class When_adding_a_parameter_value_to_a_parameter_value_cache : concern_for_ParameterValuesCache
+   {
+      private ParameterValue _parameterValue1;
+      private ParameterValue _parameterValue2;
+
+      protected override void Context()
+      {
+         base.Context();
+         _parameterValue1 = new ParameterValue("PATH1", 11, 0.1);
+         _parameterValue2 = new ParameterValue("PATH2 [mg]", 21, 0.2);
+      }
+
+      protected override void Because()
+      {
+         sut.Add(new[] {_parameterValue1, _parameterValue2});
+         sut.Add(new[] {new ParameterValue(_parameterValue1.ParameterPath, 12, 0.3), new ParameterValue(_parameterValue2.ParameterPath, 22, 0.5)});
+      }
+
+      [Observation]
+      public void should_store_the_value_for_the_given_parameter_path_and_be_able_to_retrieve_the_parameter_paths_afterwards()
+      {
+         sut.AllParameterPaths().ShouldOnlyContainInOrder(_parameterValue1.ParameterPath, _parameterValue2.ParameterPath);
+      }
+
+      [Observation]
+      public void should_return_true_if_asked_if_it_contains_some_values_for_the_added_path()
+      {
+         sut.Has(_parameterValue1.ParameterPath).ShouldBeTrue();
+         sut.Has(_parameterValue2.ParameterPath).ShouldBeTrue();
+      }
+
+
+      [Observation]
+      public void should_return_true_if_asked_if_it_contains_some_values_for_the_added_path_without_unit()
+      {
+         sut.Has("PATH2").ShouldBeTrue();
+      }
+
+      [Observation]
+      public void should_be_able_to_retrieve_the_values_for_the_individual_path()
+      {
+         sut.ValuesFor(_parameterValue1.ParameterPath).ShouldOnlyContainInOrder(11, 12);
+      }
+
+      [Observation]
+      public void should_be_able_to_retrieve_the_values_for_the_individual_path_as_array()
+      {
+         sut.GetValues(_parameterValue1.ParameterPath).ShouldOnlyContainInOrder(11, 12);
+      }
+
+      [Observation]
+      public void should_be_able_to_retrieve_the_values_for_the_individual_path_without_unit()
+      {
+         sut.GetValues("PATH2").ShouldOnlyContainInOrder(21, 22);
+      }
+   }
+
+
+   public class When_adding_a_parameter_value_with_similar_path_to_a_parameter_value_cache : concern_for_ParameterValuesCache
+   {
+      private ParameterValue _parameterValue1;
+      private ParameterValue _parameterValue2;
+
+      protected override void Context()
+      {
+         base.Context();
+         _parameterValue1 = new ParameterValue("PATH_P1", 11, 0.1);
+         _parameterValue2 = new ParameterValue("PATH_P [mg]", 21, 0.2);
```

Comment:
@Yuri05 This is your use case. Did not work at the beginning. Worked after the fix